### PR TITLE
feat: scheduled root propagation mechanism

### LINKED ISSUE #42: Develop scheduled root propagation mechanism
### Description

Create a JavaScript module that periodically calls the `propagateRoot` method on the `LineaStateBridge` contract using ethers.js and the Linea SDK. This module will automate the process of propagating roots from L1 to L2.

### Tasks

1. Add necessary dependencies to the project (ethers.js, Linea SDK)
2. Create a new JavaScript file for the purpose of this issue
3. Create a main function that periodically calls `propagateRoot` on `LineaStateBridge`
4. Implement error handling and logging for failed transactions
5. Add configuration options for period, LineaStateBridge address, and other necessary parameters
6. Set up secure environment variable handling for the private key

### Acceptance Criteria

- [ ] JavaScript module successfully calls `propagateRoot` on `LineaStateBridge` at specified intervals
- [ ] Module accepts period and LineaStateBridge address as arguments
- [ ] Private key is securely read from an environment variable
- [ ] Proper error handling and logging are implemented
- [ ] Code is well-documented and follows best practices

### Caveats

- Ensure the private key is never logged or exposed in any way
- Consider gas price fluctuations and implement a mechanism to handle failed transactions due to insufficient gas

### Resources

- [ethers.js Documentation](https://docs.ethers.org/v6/)
- [Linea SDK Documentation](https://docs.linea.build/developers/guides/linea-sdk)

### PR DESCRIPTION
**Pull request by TropicalDog17**
Closes #42 

### Example

` node scripts/propagateRoot.js --address 0x4f71bb382a119d0545c28fb532cad673af321c9f  --period 60` (60s)

image

Transaction: https://sepolia.etherscan.io/tx/0xfb448ae614973971d24ea283ee29c35b610bcd097ff0cb5dffab2dd4c7782db1

### CODE REVIEW

**Review by kfastov — CHANGES_REQUESTED** [review]
Looks legitimate, just remove the fs dependency and check if we should increase gas on the unpredictable gas error (I can be wrong at it) and we can finish with it.

File: scripts/LineaRootPropagator.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        const receipt = await tx.wait();
+        console.log(`Root propagated successfully. Transaction hash: ${receipt.transactionHash}`);
+        return;
+      } catch (error) {
+        console.error('Error propagating root:', error.message);
+        if (error.code === 'INSUFFICIENT_FUNDS') {
+          console.log('Insufficient funds for gas. Please check your wallet balance.');
+          return;
+        }
+        if (error.code === 'UNPREDICTABLE_GAS_LIMIT' || error.message.includes('transaction underpriced')) {
```
[the remark this answers is not in the document]
Comment (TropicalDog17) [reply scripts/LineaRootPropagator.js]: yeah I think I misunderstand this error, I have adjusted, please check @kfastov

**Comment by kfastov** [comment]
Thank you!

**Review by kfastov — APPROVED** [review]
I like how this PR is made

**Pull request merged**